src/main/resources/assets/opencomputers/loot/openos/lib/serialization.lua

```diff
@@ -1,134 +1,93 @@
-      limit = limit - 1
-    end
-    if truncate then
-      return result:sub(1, truncate) .. "..."
-    end
-  end
-  return result
+------------------------------------------------- Private methods -----------------------------------------------------------------
+
+local function doSerialize(array, text, prettyLook, indentationSymbol, oldIndentationSymbol, equalsSymbol)
+	text = {"{"}
+	table.insert(text, (prettyLook and "\n" or nil))
+	
+	for key, value in pairs(array) do
+		local keyType, valueType, stringValue = type(key), type(value), tostring(value)
+
+		if keyType == "number" or keyType == "string" then
+			table.insert(text, (prettyLook and indentationSymbol or nil))
+			table.insert(text, "[")
+			table.insert(text, (keyType == "string" and table.concat({"\"", key, "\""}) or key))
+			table.insert(text, "]")
+			table.insert(text, equalsSymbol)
+			
+			if valueType == "number" or valueType == "boolean" or valueType == "nil" then
+				table.insert(text, stringValue)
+			elseif valueType == "string" or valueType == "function" then
+				table.insert(text, "\"")
+				table.insert(text, stringValue)
+				table.insert(text, "\"")
+			elseif valueType == "table" then
+				table.insert(text, table.concat(doSerialize(value, text, prettyLook, table.concat({indentationSymbol, indentationSymbol}), table.concat({oldIndentationSymbol, indentationSymbol}), equalsSymbol)))
+			else
+				error("Unsupported table value type: " .. valueType)
+			end
+			
+			table.insert(text, ",")
+			table.insert(text, (prettyLook and "\n" or nil))
+		else
+			error("Unsupported table key type: " .. keyType)
+		end
+	end
+
+	table.remove(text, (prettyLook and #text - 1 or #text))
+	table.insert(text, (prettyLook and oldIndentationSymbol or nil))
+	table.insert(text, "}")
+	return text
+end
+
+------------------------------------------------- Public methods -----------------------------------------------------------------
+
+function serialization.serialize(array, prettyLook, indentationWidth, indentUsingTabs)
+	checkArg(1, array, "table")
+	indentationWidth = indentationWidth or 2
+	local indentationSymbol = indentUsingTabs and "	" or " "
+	indentationSymbol, indentationSymbolHalf = string.rep(indentationSymbol, indentationWidth)
+	return table.concat(doSerialize(array, {}, prettyLook, indentationSymbol, "", prettyLook and " = " or "="))
+end
+
+function serialization.unserialize(serializedString)
+	checkArg(1, serializedString, "string")
+	local success, result = pcall(load("return " .. serializedString))
+	if success then return result else return nil, result end
+end
+
+function serialization.serializeToFile(path, array, prettyLook, indentationWidth, indentUsingTabs, appendToFile)
+	checkArg(1, path, "string")
+	checkArg(2, array, "table")
+	filesystem.makeDirectory(filesystem.path(path) or "")
+	local file = io.open(path, appendToFile and "a" or "w")
+	file:write(serialization.serialize(array, prettyLook, indentationWidth, indentUsingTabs))
+	file:close()
 end
```

Comment:
path shouldn't be the first argument, but given the optional trailing arguments, I understand the predicament. There are few things to consider with this method. Because you are building the entire string (doSerialize) and  then writing to file, you are not saving the user any memory or performance costs. It would be best to let the filesystem handle the buffering and write one element of the table at a time. Consider having doSerialize return the table of strings, and write to file one segment at a time. With that change, you can have serialize concat the table itself before returning.
Also, this method shouldn't be trying to share feature space with open mode (i.e. you're asking for the open mode). The user should just pass the filehandle.
serialization.serialize(value, pretty, options)
...
```
if options.output then
  for _,segment in ipairs(result) do
    options.output:write(segment)
  end
else
  return table.concat(result)
end
```

I need to give this more thought. I'm not sure I have the best solution yet.